irc/server.go

```diff
@@ -661,39 +663,17 @@ func (server *Server) applyConfig(config *Config, initial bool) (err error) {
 		server.channels.loadRegisteredChannels()
 	}
 
-	// MaxLine
-	if config.Limits.LineLen.Rest != 512 {
-		SupportedCapabilities.Enable(caps.MaxLine)
-		value := fmt.Sprintf("%d", config.Limits.LineLen.Rest)
-		CapValues.Set(caps.MaxLine, value)
-	} else {
-		SupportedCapabilities.Disable(caps.MaxLine)
-	}
-
 	// STS
 	stsPreviouslyEnabled := oldConfig != nil && oldConfig.Server.STS.Enabled
-	stsValue := config.Server.STS.Value()
-	stsDisabledByRehash := false
-	stsCurrentCapValue, _ := CapValues.Get(caps.STS)
+	stsValue := config.Server.capValues[caps.STS]
+	stsCurrentCapValue := ""
+	if oldConfig != nil {
+		stsCurrentCapValue = oldConfig.Server.capValues[caps.STS]
+	}
 	server.logger.Debug("server", "STS Vals", stsCurrentCapValue, stsValue, fmt.Sprintf("server[%v] config[%v]", stsPreviouslyEnabled, config.Server.STS.Enabled))
-	if config.Server.STS.Enabled {
-		// enabling STS
-		SupportedCapabilities.Enable(caps.STS)
-		if !stsPreviouslyEnabled {
-			addedCaps.Add(caps.STS)
-			CapValues.Set(caps.STS, stsValue)
-		} else if stsValue != stsCurrentCapValue {
-			// STS policy updated
-			CapValues.Set(caps.STS, stsValue)
-			updatedCaps.Add(caps.STS)
-		}
-	} else {
-		// disabling STS
-		SupportedCapabilities.Disable(caps.STS)
-		if stsPreviouslyEnabled {
-			removedCaps.Add(caps.STS)
-			stsDisabledByRehash = true
-		}
+	if (config.Server.STS.Enabled != stsPreviouslyEnabled) || (stsValue != stsCurrentCapValue) {
+		// XXX: STS is always removed by CAP NEW sts=duration=0, not CAP DEL
```

Comment:
Not a fan of `XXX` comments, personally. I'm left unsure whether this is a TODO that needs to be implemented or just a note about the weird behaviour.

I think in this case it's a note? It's a bit hard to follow how this works - i.e., how the cap value ends up as `duration=0` if STS is disabled. A slightly longer comment would be nice :)

Reply:
It's a note. I like `XXX` a lot as a convention :-) I'll do a longer comment to go with it.